oreo.go

```diff
@@ -218,6 +220,28 @@ func (c *Client) saveCookies(resp *http.Response) error {
 		cookies = mergedCookies
 	}
 
+	lockFile := fmt.Sprintf("%s.lock", c.cookieFile)
+	lock := flock.NewFlock(lockFile)
+	locked := false
+	for i := 0; i < 10; i++ {
```

Comment:
yeah, I suppose.  I will create a CookieSaveError struct so we can return a known type for this error condition and users can choose to ignore this if they want.